src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -202,6 +202,12 @@ private static Assembly ResolveForUnknownRequestor(string requestedAssemblyName)
                     foreach (string loadedAssemblyFullPath in s_assembliesFromFiles.Keys)
                     {
                         string directoryPath = Path.GetDirectoryName(loadedAssemblyFullPath);
+
+                        if (!string.IsNullOrEmpty(requestedAssemblyIdentity.CultureName))
+                        {
+                            directoryPath = Path.Combine(directoryPath, requestedAssemblyIdentity.CultureName);
```

Comment:
No, it will not automatically look in a 1041 directory. The probing logic is described here:
https://msdn.microsoft.com/en-us/library/15hyw9x3(v=vs.100).aspx